src/main/java/Duke.java

```diff
@@ -1,10 +1,112 @@
+import java.lang.*;
+import java.util.*;
+import java.io.File;
+
 public class Duke {
+    static String indent = "     ";
+    static String line = "    ____________________________________________________________";
+    private static final String FILEPATH = ""
+
+    public static void updateTask(Task task) {
+        try {
+            FileOutputStream f = new FileOutputStream(new File("tasks.json"));
+            ObjectOutputStream o = new ObjectOutputStream(f);
+
+            // Write updated tasks to file
+            o.writeObject(p1);
+
+            o.close();
+            f.close();
+
+            FileInputStream fi = new FileInputStream(new File("tasks.json"));
+            ObjectInputStream oi = new ObjectInputStream(fi);
+
+            // Read objects
+            Task t1 = (Task) oi.readObject();
+
+            oi.close();
+            fi.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found");
+        } catch (IOException e) {
+            System.out.println("Error initializing stream");
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        // Greeting
+        System.out.println(line);
+        System.out.println(indent+"Hello! I'm Duke\n"+
+                indent+
+                "What can I do for you?");
+        System.out.println(line);
+        System.out.println();
+
+        // Data Structures
+        ArrayList<Task> commands = new ArrayList<>();
+
+        Scanner in = new Scanner(System.in);
+        String input = in.nextLine();
+
+        while (!input.equals("bye")) {
+            StringTokenizer st = new StringTokenizer(input);
+            System.out.println(line);
+            String cmd = st.nextToken();
+
+            if (cmd.equals("list")) {
+
+                System.out.println(indent+"Here are the tasks in your list:");
+                for (int i=0; i<commands.size(); i++) {
+                    System.out.println(indent+(i+1)+"."+commands.get(i));
+                }
+
+            } else if (cmd.equals("mark") || cmd.equals("unmark")) {
+
+                int index = Integer.parseInt(st.nextToken());
+                commands.get(index-1).setDone(cmd.equals("mark"));
+                if (cmd.equals("mark")) {
+                    System.out.println(indent + "Nice! I've marked this task as done:");
+                } else {
+                    System.out.println(indent + "OK, I've marked this task as not done yet:");
+                }
+                System.out.println(indent + "  " + commands.get(index-1));
+
```

Comment:
Maybe remove the Overarching else if check, as there is a nested check? Makes code seem messy.